src/components/grid-view/GridView.js

```diff
@@ -38,33 +39,43 @@ class GridView extends React.Component<Props> {
     });
 
     componentDidUpdate(prevProps: Props) {
-        const { columnCount, width } = this.props;
+        const { columnCount, currentCollection, width } = this.props;
 
         // The React Virtualized Table must be notified when either the cached
         // row sizes or the parent width change. If omitted, rows are sized
         // incorrectly resulting in gaps or content overlap.
-        if (columnCount !== prevProps.columnCount || width !== prevProps.width) {
+        if (
+            columnCount !== prevProps.columnCount ||
+            currentCollection.id !== prevProps.currentCollection.id ||
+            width !== prevProps.width
+        ) {
             this.cache.clearAll();
             this.forceUpdate();
         }
     }
 
     cellRenderer = ({ dataKey, parent, rowIndex }: TableCellRendererParams) => {
-        const { columnCount, currentCollection, slotRenderer } = this.props;
+        const { columnCount, currentCollection, onItemSelect, slotRenderer } = this.props;
         const count = getProp(currentCollection, 'items.length', 0);
         const contents = [];
 
         const startingIndex = rowIndex * columnCount;
         const maxSlotIndex = Math.min(startingIndex + columnCount, count);
 
         for (let slotIndex = startingIndex; slotIndex < maxSlotIndex; slotIndex += 1) {
+            const item = getProp(currentCollection, `items[${slotIndex}]`);
+
             // using item's id as key is important for renrendering.  React Virtualized Table rerenders
             // on every 1px scroll, so using improper key would lead to image flickering in each
             // card of the grid view when scrolling.
             contents.push(
                 <div
-                    key={getProp(currentCollection, `items[${slotIndex}].id`) || uniqueId('bdl-GridView-slot')}
-                    className="bdl-GridView-slot"
+                    key={getProp(item, `id`) || uniqueId('bdl-GridView-slot')}
+                    className={classNames('bdl-GridView-slot', {
+                        'bdl-GridView-slot--selected': getProp(item, 'selected'),
+                    })}
+                    onClick={() => onItemSelect(item)}
+                    role="presentation"
```

Comment:
A few times I ran into the es-lint rule: `Visible, non-interactive elements with click handlers must have at least one keyboard listener.` . Putting `role="presentation` "fixed" the problem.  Is this ok, or am I dodging the problem?

Reply:
Yeah that seems like dodging the problem. Seems like `role="presentation"` is to remove semantic meaning from the element (https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/presentation/). It makes more sense to allow keyboard users to hit "Enter" and "Space" to trigger the same action as click